sdk/storage/Azure.Storage.DataMovement.Blobs/src/DataMovementBlobsExtensions.cs

```diff
@@ -637,34 +638,50 @@ internal static StorageResourceItemProperties ToResourceProperties(this BlobItem
                 properties: properties);
         }
 
+        private static string ConvertContentPropertyObjectToString(object contentPropertyValue)
+        {
+            if (contentPropertyValue is string)
+            {
+                return Convert.ToString(contentPropertyValue);
+            }
+            else if (contentPropertyValue is string[])
+            {
+                return string.Join(",", (string[])contentPropertyValue);
+            }
+            else
+            {
+                throw Errors.UnexpectedPropertyType(DataMovementConstants.ResourceProperties.ContentType, DataMovementConstants.StringTypeStr, DataMovementConstants.StringArrayTypeStr);
```

Comment:
This isn't always `DataMovementConstants.ResourceProperties.ContentType` right? I guess we could take this in?